contracts/sol6/KyberNetwork.sol

```diff
@@ -571,27 +571,23 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
             return true;
         }
 
-        ReservesData memory reservesData = (src == ETH_TOKEN_ADDRESS)
-            ? tradeData.ethToToken
-            : tradeData.tokenToEth;
-        uint256 callValue;
+        uint256 actualDestAmount;
+        uint256 srcDecimals = src == ETH_TOKEN_ADDRESS ? ETH_DECIMALS : reservesData.decimals;
+        uint256 destDecimals = src == ETH_TOKEN_ADDRESS ? reservesData.decimals : ETH_DECIMALS;
 
         for (uint256 i = 0; i < reservesData.addresses.length; i++) {
-            callValue = (src == ETH_TOKEN_ADDRESS) ? reservesData.srcAmounts[i] : 0;
-
-            // reserve sends tokens/eth to network. network sends it to destination
-            require(
-                reservesData.addresses[i].trade{value: callValue}(
-                    src,
-                    reservesData.srcAmounts[i],
-                    dest,
-                    address(this),
-                    reservesData.rates[i],
-                    true
-                ),
-                "trade failed"
+            actualDestAmount += tradeAndVerifyNetworkBalance(
+                reservesData.addresses[i],
+                src,
+                reservesData.srcAmounts[i],
+                dest,
+                reservesData.rates[i],
+                srcDecimals,
+                destDecimals
             );
         }
+        // verify balance after all reserve trades
+        require(actualDestAmount >= expectedDestAmount, "amt low");
```

Comment:
Can we avoid the actualDestAmount calculation and compare?
since if each reserve sends correct amount. it seems like extra check

Reply:
Yes, it is removed, will push the changes later along with some tests